scripts/o3de/o3de/engine_template.py

```diff
@@ -2119,6 +2121,15 @@ def create_gem(gem_path: pathlib.Path,
     # remove the first and last quote because those already exist in gem.json
     replacements.append(("${UserTags}", tags_quoted[1:-1]))
 
+    temp_platforms = []
+    if platforms:
+        #same as tags, allow commas or spaces as platform separators
+        new_platforms = platforms.replace(',', ' ').split() if isinstance(platforms, str) else platforms
+        temp_platforms.extend(new_platforms)
+    platforms_quoted = ','.join(f'"{word.strip()}"' for word in set(temp_platforms))
```

Comment:
fyi: set is unordered and could change from the order the user has entered in the platforms to the CLI.
You can use `dict.fromkeys(temp_platforms)` to filter out duplicates, but maintain the insertion order.

See this stackoverflow question for more info: https://stackoverflow.com/questions/9792664/converting-a-list-to-a-set-changes-element-order